File: frontend/src/app/protected/services/user.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {User} from "../classes/User";

@Injectable({
  providedIn: 'root'
})
export class UserService {

  constructor(private http: HttpClient) {

  }

  getAllUsers(): Observable<User[]> {
    return this.http.get<User[]>("api/users/getAllUsers");
  }

  activateUser(userId: number): Observable<User> {
    return this.http.put<User>(`api/users/activateUser/${userId}`, null);
  }

  deactivateUser(userId: number): Observable<User> {
    return this.http.put<User>(`api/users/deactivateUser/${userId}`, null);
  }

  changeToAdmin(userId: number): Observable<User> {
    return this.http.put<User>(`api/users/changeToAdmin/${userId}`, null);
  }

  changeToUser(userId: number): Observable<User> {
    return this.http.put<User>(`api/users/changeToUser/${userId}`, null);
  }

  deleteUserById(userId: number) {
    return this.http.delete(`api/users/deleteById/${userId}`);
  }
}

File: backend/src/main/java/ut/ee/hades/app/web/services/UserService.java
package ut.ee.hades.app.web.services;

import ut.ee.hades.app.web.model.dto.UserDTO;

import java.util.List;

public interface UserService {

    /**
     * Retrieves all system users
     * @return - all users
     */
    List<UserDTO> getAllUsers();


    /**
     * Activates registered user
     * @param userId - target user id
     * @return - user with status set as activated
     */
    UserDTO activateUser(Long userId);


    /**
     * Deactivates system user
     * @param userId - target user id
     * @return - user with status set as registered
     */
    UserDTO deactivateUser(Long userId);


    /**
     * Changes user to system admin
     * @param userId - target user id
     * @return - user set as admin
     */
    UserDTO changeToAdmin(Long userId);


    /**
     * Changes user to regular system user
     * @param userId - target user id
     * @return - user set to regular user
     */
    UserDTO changeToUser(Long userId);


    /**
     * Deletes the user with given id
     * @param userId - user id
     */
    void deleteUserById(Long userId);
}
